packages/@glimmer/runtime/lib/compiled/opcodes/dom.ts

```diff
@@ -237,27 +238,34 @@ APPEND_OPCODES.add(Op.DynamicAttr, (vm, { op1: _name, op2: _trusting, op3: _name
   let attribute = vm.elements().setDynamicAttribute(name, value, trusting, namespace);
 
   if (!isConstRef(reference)) {
-    vm.updateWith(new UpdateDynamicAttributeOpcode(reference, attribute));
+    vm.updateWith(new UpdateDynamicAttributeOpcode(reference, attribute, vm.env));
   }
 });
 
 export class UpdateDynamicAttributeOpcode extends UpdatingOpcode {
   public type = 'patch-element';
 
-  public lastValue: unknown;
+  private updateRef: Reference;
 
-  constructor(private reference: Reference<unknown>, private attribute: DynamicAttribute) {
+  constructor(reference: Reference<unknown>, attribute: DynamicAttribute, env: Environment) {
     super();
-    this.lastValue = valueForRef(reference);
-  }
 
-  evaluate(vm: UpdatingVM) {
-    let { attribute, reference, lastValue } = this;
-    let currentValue = valueForRef(reference);
+    let initialized = false;
 
-    if (currentValue !== lastValue) {
-      attribute.update(currentValue, vm.env);
-      this.lastValue = currentValue;
-    }
+    this.updateRef = createComputeRef(() => {
+      let value = valueForRef(reference);
+
+      if (initialized === true) {
+        attribute.update(value, env);
```

Comment:
FWIW, I still don't think this guarantees that the `DynamicAttribute` will actually update things. Some of the subclasses of `DynamicAttribute` have caches that do not check if the DOM value was changed out from under them.

For example:

https://github.com/glimmerjs/glimmer-vm/blob/4b697982032bdf6bc1fdbcb2e427f8d50fa07e1b/packages/%40glimmer/runtime/lib/vm/attributes/dynamic.ts#L117-L127

The reason the test added above passes is because `input.value` has a custom `DynamicAttribute` subclass (that doesn't do the last value caching that `DefaultDynamicProperty` does):

https://github.com/glimmerjs/glimmer-vm/blob/4b697982032bdf6bc1fdbcb2e427f8d50fa07e1b/packages/%40glimmer/runtime/lib/vm/attributes/dynamic.ts#L175-L182

Reply:
Note: this almost certainly does address the reported issue, it just doesn't 100% fix the full category of issues. I'm fairly sure you could end up with this same issue with other element + attribute combinations that the user could mutate. We have custom overrides for a few common ones (`input.value`, `textarea.value`, `option.selected`), but that list of overrides is not exhaustive (e.g. video element has some local mutable attributes | properties that could have been changed by the user).